rclcpp/include/rclcpp/create_subscription.hpp

```diff
@@ -136,8 +137,9 @@ create_subscription(
     topic_name,
     factory,
     options.template to_rcl_subscription_options<MessageT>(qos),
-    use_intra_process);
-  node_topics->add_subscription(sub, options.callback_group);
+    use_intra_process,
+    options.intra_process_buffer_type);
```

Comment:
Do I care about setting `options.intra_process_buffer_type` if `use_intra_process` is false?

Reply:
No, `intra_process_buffer_type` is used only when `use_intra_process` is true.
I also don't really like it, but the alternative I tried was to split this into 2 functions:
```
create_subscription()
if (use_intra_process)
    create_intra_process_subscription(options.intra_process_buffer_type)
```